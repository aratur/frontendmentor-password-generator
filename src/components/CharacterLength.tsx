import React from 'react';
import Box from '@mui/material/Box';
import Typography from '@mui/material/Typography';
import useMediaHook from '../hooks/mediaHook';

type Props = {
  charLength: number;
  label: string;
};

const CharacterLength = (props: Props) => {
  const { charLength, label } = props;
  const media = useMediaHook();
  return (
    <Box
      sx={{
        display: 'flex',
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center',
        height: '43px',
      }}
    >
      <Typography variant={media ? 'body2' : 'body1'} component="p">
        {label}
      </Typography>
      <Typography variant={media ? 'h1' : 'h2'} component="p" color="primary">
        {charLength}
      </Typography>
    </Box>
  );
};

export default CharacterLength;
