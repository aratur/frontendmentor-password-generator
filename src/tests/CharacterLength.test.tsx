import React from 'react';
import { screen, render } from '@testing-library/react';
import CharacterLength from '../components/CharacterLength';

describe('Character Length', () => {
  it('snapshot', () => {
    const { container } = render(
      <CharacterLength label="Character Length" charLength={1234} />
    );
    expect(container).toMatchSnapshot();
  });
  it('renders props', () => {
    render(<CharacterLength label="Character Length" charLength={1234} />);
    const chars = screen.getByText(/1234/i);
    expect(chars).toBeInTheDocument();
  });
});
