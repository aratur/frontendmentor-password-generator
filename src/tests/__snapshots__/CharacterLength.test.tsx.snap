// Vitest Snapshot v1

exports[`Character Length > snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-4rmty7"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      Character Length
    </p>
    <p
      class="MuiTypography-root MuiTypography-h2 css-wf8t3v-MuiTypography-root"
    >
      1234
    </p>
  </div>
</div>
`;
